subprojects/maven/src/main/java/org/gradle/api/publish/maven/internal/publication/MavenPomDependencies.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.publish.maven.internal.publication;
+
+import org.gradle.api.publish.maven.MavenDependency;
+import org.gradle.api.publish.maven.internal.dependencies.MavenDependencyInternal;
+
+import java.util.Set;
+
+/**
+ * The dependencies of a Maven POM.
+ * <p>
+ * TODO: Currently this represents a POM's dependencies in a "Gradle" point of view,
+ * since the actual separation of maven scopes is implemented by separating dependencies
+ * of each scope into separate collections. The same applies for optional dependencies.
+ * <p>
+ * To better align this interface with the way Maven represents dependencies in a POM,
```

Comment:
🤔 We could also keep this internal representation and add a default `List<MavenDependency> getMavenDependenciesView()` method here to transform this into that.  I wonder if it might be better to keep this internal structure as is.